app/src/main/java/com/alexchan/wallpaper/ui/MainActivity.kt

```diff
@@ -0,0 +1,172 @@
+                            Toast.makeText(this@MainActivity, "Searching: $newText", Toast.LENGTH_SHORT).show()
+
+                            searchQuery = newText
+                            searchStatus = true
+
+                            val navHostFragment = supportFragmentManager.findFragmentById(R.id.mainNavHostFragment)
+                            val navController = navHostFragment?.findNavController()
+                            navController?.navigate(R.id.wallpaperFragment)
+                        }
+                        return true
+                    }
+                })
+                true
+            }
+            R.id.listView -> {
+                // Handle List View
+                //photosGrid.layoutManager = LinearLayoutManager(this, LinearLayoutManager.VERTICAL, false)
+                photosGrid.layoutManager = StaggeredGridLayoutManager(1, StaggeredGridLayoutManager.VERTICAL)
+                true
+            }
+            R.id.gridView -> {
+                // Handle Grid View
+                //unsplashPhotoImageView.scaleType = ImageView.ScaleType.CENTER_CROP
+                //unsplashPhotoImageView.requestLayout()
+                photosGrid.layoutManager = GridLayoutManager(this, 2)
+                true
+            }
+            R.id.staggeredGridView -> {
+                // Handle Staggered View
+                photosGrid.layoutManager = StaggeredGridLayoutManager(2, StaggeredGridLayoutManager.VERTICAL)
+                true
+            }
+            else -> false
+        }
+    }
+
+    // Open Navigation Drawer
+    private fun openNavDrawer() {
+        updateNavigationDrawerSetCheckedItem()
+        Log.d(TAG, "Drawer is Open")
+        drawerLayout.openDrawer(GravityCompat.START)
+    }
+
+    // Update on Swipe Open Navigation Drawer Set Checked Item State
+    private fun updateOnSwipeOpenNavDrawer() {
+        drawerLayout.addDrawerListener(object : DrawerLayout.SimpleDrawerListener() {
+            override fun onDrawerSlide(drawerView: View, slideOffset: Float) {
+                super.onDrawerSlide(drawerView, slideOffset)
+                updateNavigationDrawerSetCheckedItem()
+            }
+        })
+    }
+
+    // Update Navigation Drawer Set Checked Item State
+    private fun updateNavigationDrawerSetCheckedItem() {
+        Log.d(TAG, supportFragmentManager.findFragmentById(R.id.mainNavHostFragment)
+            ?.findNavController()?.currentDestination?.id.toString())
+        when (supportFragmentManager.findFragmentById(R.id.mainNavHostFragment)?.findNavController()?.currentDestination?.id) {
+            R.id.wallpaperFragment -> navigationView.setCheckedItem(R.id.wallpaperFragment)
+            R.id.notificationFragment -> navigationView.setCheckedItem(R.id.notificationFragment)
+            R.id.profileFragment -> navigationView.setCheckedItem(R.id.profileFragment)
+        }
+    }
+
+    override fun onBackPressed() {
```

Comment:
This is actually bad. If you have a toolbar in the details screen, try this `toolbar.setupWithNavController(findNavController())`